### DISCUSSION TITLE
`.chezmoiroot` and location of other `.chezmoi*` files

### DISCUSSION THREAD

**Discussion by 2-4601** [twpayne/chezmoi#3109] [Q&A] [answered]
I was planning to start using [`.chezmoiroot`](https://www.chezmoi.io/user-guide/advanced/customize-your-source-directory/#use-a-subdirectory-of-your-dotfiles-repo-as-the-root-of-the-source-state) and move everything to a separate `home/` directory.

The docs instruct to _"move the relevant files in to the new root subdirectory manually."_ So I created `.chezmoiroot` and moved all the files and directories (except `.chezmoi*`) to `home/`.

Then I ran `chezmoi apply` and got an error about a missing template. So I moved `.chezmoitemplates/` to `home/` and run `chezmoi apply` again. The missing template error was fixed but I run into other issues. After several trials and errors it turned out that only `.chezmoiroot` and `.chezmoiversion` can live outside the `home/`.

So this is what seems to be working:
```
.
├── .chezmoiroot
├── .chezmoiversion
└── home/
    ├── .chezmoiexternal.toml
    ├── .chezmoiignore
    ├── .chezmoiscripts/
    ├── .chezmoitemplates/
    ├── .chezmoi.toml.tmpl
    ├── ...
    ...
```
This is the layout I was aiming for:
```
.
├── .chezmoiexternal.toml
├── .chezmoiignore
├── .chezmoiroot
├── .chezmoiscripts/
├── .chezmoitemplates/
├── .chezmoiversion
├── .chezmoi.toml.tmpl
└── home/
    ├── ...
    ...
```
Is it possible to make this layout work?

**Comment by halostatue** [comment, marked as answer]
You could probably only do so with symlinks due to the way that `.chezmoiroot` works. Basically, when running, chezmoi looks at `.chezmoiversion` *first* so that it can error out if there is a version mismatch. It then looks at `.chezmoiroot` and then changes its "base" directory from `.chezmoi.workingTree` to the directory specified in `.chezmoiroot`. During most operations, chezmoi would then no longer "see" anything in `.chezmoi.workingTree`. In effect, it performs `cd (cat .chezmoiroot)`.
